packages/vite/src/node/server/transformRequest.ts

```diff
@@ -271,6 +271,23 @@ async function loadAndTransform(
     if (map.mappings && !map.sourcesContent) {
       await injectSourcesContent(map, mod.file, logger)
     }
+    for (
+      let sourcesIndex = 0;
+      sourcesIndex < map.sources.length;
+      ++sourcesIndex
+    ) {
+      const sourcePath = map.sources[sourcesIndex]
+
+      // Rewrite sources to relative paths to give debuggers the chance
+      // to resolve and display them in a meaningful way (rather than
+      // with absolute paths).
+      if (path.isAbsolute(sourcePath) && path.isAbsolute(mod.file)) {
+        map.sources[sourcesIndex] = path.relative(
+          path.dirname(mod.file),
+          sourcePath,
+        )
```

Comment:
@sapphi-red just checking here that this `path.relative` call won't be affected by issues like the one you were worried about with [paths in different drives on windows](https://github.com/vitejs/vite/pull/7173#discussion_r825380501) that can't be expressed as relative paths. I assume the path would be returned as absolute in this case. To be honest, maybe we should document having Vite work across hard drives as out of scope if this will increase complexity for everyone else.

Reply:
Windows also has this concept of relative paths with drive letters, which I think would also break vite. 

I suppose instead of using `path.relative` here, we could also do string gymnastics ourselves, but that likely would be even worse / unpredictable.

Reply:
> I assume the path would be returned as absolute in this case.

Yes, you're correct. I think we can ignore the cross drive usage as it doesn't work in many places already at least for now.